### DISCUSSION TITLE
JavaScript LSP lags behind actual code (shows errors where there are none)

### DISCUSSION THREAD

**Discussion by aral** [helix-editor/helix#10639] [Troubleshooting]
### Summary

Sometimes, when editing JavaScript, the LSP will start lagging behind the actual code. Once that happens the only thing that fixes it is to restart Helix Editor.

## Screenshot

![Screenshot from 2022-08-31 15-27-44](https://user-images.githubusercontent.com/41798/187705314-0ab28be9-502a-4322-b170-0e76f6285fd7.png)

### Reproduction Steps

I don’t know how to reproduce this or what causes it.

### Helix log

The only entry I have from today is from an hour ago so I’m assuming it’s not related (unless Helix’s timestamps don’t take daylight savings time into consideration):

```
2022-08-31T14:25:54.503 helix_term::application [WARN] lsp position out of bounds - Diagnostic { range: Range { start: Position { line: 210, character: 0 }, end: Position { line: 210, character: 1 } }, severity: Some(Error), code: Some(Number(1109)), code_description: None, source: Some("typescript"), message: "Expression expected.", related_information: None, tags: None, data: None }
```

### Platform

Linux (Fedora Silverblue 36)

### Terminal Emulator

Black Box

### Helix Version

helix 22.05 (f0fb3407)

**Comment by David-Else** [comment]
Could be related to my `lsp out of bounds` problems, which are also only manifesting today: https://github.com/helix-editor/helix/issues/3617

**Comment by EmmChriss** [comment]
I've been having the same problem

**Comment by pedronasser** [comment]
I honestly feel that this is not just related to Javascript but any LSP integration.
I've had a similar thing with rust-analyzer.

**Comment by iljarotar** [comment]
I'm having the same problem quite often unfortunately. But instead of restarting helix, I have found that deleting the lines responsible for the error and then undoing the deletion fixes the error message. Not sure if it always works, though.
